modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -3001,6 +3008,28 @@ public IgniteConfiguration setAuthenticationEnabled(boolean authEnabled) {
         return this;
     }
 
+    /**
+     * Returns hard timeout for partition map exchange process. If it doesn't finish until the specified amount of time
+     * passes, all nodes that potentially could block progress will be removed from cluster.
+     *
+     * @return Hard exchange timeout, in millis.
+     */
+    public long getExchangeHardTimeout() {
+        return exchangeHardTimeout;
+    }
+
+    /**
+     * Sets hard timeout for partition map exchange process.
+     *
+     * @param timeout Hard exchange timeout, in millis.
+     * @return {@code this} for chaining.
+     */
+    public IgniteConfiguration setExchangeHardTimeout(long timeout) {
```

Comment:
What should I do, if I don't want this functionality? Should I set this timeout to Long.MAX_VALUE? It will make us create a lot of schedules, that will never be executed. I think, -1 should be a special value, that disables this feature.

Reply:
OK, good point.